optimizer/lib/transformers/AddBlurryImagePlaceholders.js

```diff
@@ -54,18 +54,20 @@ class AddBlurryImagePlaceholders {
       if (tagName === 'amp-video' && node.attribs.poster) {
         src = node.attribs.poster;
       }
-      if (currPlaceholderCount < MAX_BLURRED_PLACEHOLDERS && src &&
-        !this.hasPlaceholder_(node) &&
-          this.shouldAddBlurryPlaceholder_(node, src, tagName)) {
-            promises.push(this.addBlurryPlaceholder_(tree, src)
-              .then((imgChild) => {
-                if (currPlaceholderCount < MAX_BLURRED_PLACEHOLDERS) {
-                  node.appendChild(imgChild);
-                  currPlaceholderCount++;
-                }
-              })
-            );
-          }
+      if (currPlaceholderCount >= MAX_BLURRED_PLACEHOLDERS) {
+        return;
```

Comment:
Ups, my mistake. We need to `break` here as we need to return the promises. Sorry!

Reply:
No worries :)